### DISCUSSION TITLE
new error with  qiimed2meco

### DISCUSSION THREAD

**Discussion by HCB-M2RG** [ChiLiubio/file2meco#20] [Q&A]
Hello,
I am getting an error that I have never seen before after using this command to pull data into the microeco package for a while now.

the error is: error in h(simpleError(msg, call)) :
error in evaluating the argument 'x' in selecting a method for function 'as.data.frame': zip file '../Data/qiime.outputs/16S/6_denoise_dada2_table.qza' cannot be opened

when I run it through the function that I have been successfully using qimmed2meco. Here is my code block:
```{r qiime2 16S data}
qiimed2meco <- function(ASV_data, sample_data, taxonomy_data, phylo_tree = NULL){
	# Read ASV data
	ASV <- as.data.frame(read_qza(ASV_data)$data)
	#  Read metadata
	metadata <- read_q2metadata(sample_data)
	rownames(metadata) <- as.character(metadata[, 1])
	# Read taxonomy table
	taxa_table <- read_qza(taxonomy_data)
	taxa_table <- parse_taxonomy(taxa_table$data)
	# Make the taxonomic table clean, this is very important.
	taxa_table %<>% tidy_taxonomy
	# Read phylo tree
	if(!is.null(phylo_tree)){
		phylo_tree <- read_qza(phylo_tree)$data
	}
	dataset_16S <- microtable$new(sample_table = metadata, tax_table = taxa_table, otu_table = ASV, phylo_tree = phylo_tree)
	dataset_16S
}
# to test this... first download QIIME2 files from https://docs.qiime2.org/2020.8/tutorials/pd-mice/

meco_dataset_16S <- qiimed2meco(ASV_data = file.path("../Data/qiime.outputs/16S/6_denoise_dada2_table.qza"), sample_data = file.path("../Data/qiime.outputs/16S/metadata.tsv"), taxonomy_data = file.path("../Data/qiime.outputs/16S/14_Unfiltered_ASV_taxonomy.qza"), phylo_tree = file.path("../Data/qiime.outputs/16S/27_rooted_tree.qza"))
meco_dataset_16S
#note: that I changed the name of sample.metadata.tsv from the tutorials recommended sample-metadata.tsv
```

I see now in the tutorial that there is a function qiime2meco and I tried that too but got another error with respect to how the data is zipped.

Here is the error: Error in unzip(file, exdir = tmp, list = TRUE) :
zip file '../Data/qiime.outputs/16S/6_denoise_dada2_table.qza' cannot be opened

Here is the newer code block:
```{r newqimme2 16S data}
tree_data <- file.path("../Data/qiime.outputs/16S/27_rooted_tree.qza")
# please download fasta from 
rep_data <- file.path("../Data/qiime.outputs/16S/6_denoise_dada2_table.qza")
meco_dataset_16S <- qiime2meco(file.path("../Data/qiime.outputs/16S/6_denoise_dada2_table.qza"), sample_table = file.path("../Data/qiime.outputs/16S/metadata.tsv"), taxonomy_table = file.path("../Data/qiime.outputs/16S/14_Unfiltered_ASV_taxonomy.qza"), phylo_tree = tree_data, rep_fasta = rep_data, auto_tidy = TRUE)
```

this must be a problem with the file .qza file(s) that I am trying to open?? or ?? Can someone help??

**Comment by ChiLiubio** [comment]
When you run `file.path("../Data/qiime.outputs/16S/6_denoise_dada2_table.qza")` in R, is it ok to show the path? Please first make sure the path is normal. If the error is still there. Please send me your "6_denoise_dada2_table.qza" and others to have a try.

**Comment by HCB-M2RG** [comment]
﻿Hello,
I got it working but thank you for your responsiveness.
-h

Hans C. Bernstein
Professor
Microalgae & Microbiomes Research Group Leader
Arctic Centre for Sustainable Energy (ARC)
UiT - The Arctic University of Norway
Norwegian College of Fisheries Sciences
Faculty of Biosciences, Fisheries and Economics
Email: ***@***.******@***.***>
URL: https://uit.no/research/micro
Twitter: @microbe_angler<https://twitter.com/microbe_angler?ref_src=twsrc%5Egoogle%7Ctwcamp%5Eserp%7Ctwgr%5Eauthor>


From: Chi Liu ***@***.***>
Date: Wednesday, 12 April 2023 at 13:44
To: ChiLiubio/file2meco ***@***.***>
Cc: Hans Christopher Bernstein ***@***.***>, Author ***@***.***>
Subject: Re: [ChiLiubio/file2meco] new error with qiimed2meco (Discussion #20)

When you run file.path("../Data/qiime.outputs/16S/6_denoise_dada2_table.qza") in R, is it ok to show the path? Please first make sure the path is normal. If the error is still there. Please send me your "6_denoise_dada2_table.qza" and others to have a try.
